.github/workflows/csharp-sdk-tests.yml

```diff
@@ -0,0 +1,47 @@
+name: csharp-sdk
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  test-csharp-sdk:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        bun-version: [latest]
+        dotnet-version: [8.0.x]
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Bun
+        uses: oven-sh/setup-bun@v2
+        with:
+          bun-version: ${{ matrix.bun-version }}
+
+      - name: Setup .NET ${{ matrix.dotnet-version }}
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: ${{ matrix.dotnet-version }}
+
+      - name: Install dependencies
+        run: bun install --frozen-lockfile
+
+      - name: Test csharp-sdk
+        run: |
+          make test-csharp-sdk
```

Comment:
Add step (I think better inside makefile) to check correctness of target code for at least types. E.g.: compile program like: https://github.com/atomic-ehr/codegen/blob/main/examples/typescript-r4/demo.ts